docs/source/troubleshooting.rst

```diff
@@ -123,6 +123,17 @@ Sample code to check SDK WPA3 support at compile time:
     #warning "No WPA3 support."
     #endif
 
+Serial not printing
+-------------------
+I have uploaded firmware to the ESP32 device, but I don't see any response from a Serial.print (HardwareSerial).
+
+Solution
+^^^^^^^^
+
+Newer ESP32 variants have two possible USB connectors- USB and UART.  The UART connector will go through a USB->UART adapter, and will typically present itself with the name of that mfr (eg, Silicon Labs CP210x UART Bridge).  The USB connector can be used as a USB-CDC bridge and will appear as an Espressif device (Espressif USB JTAG/serial debug unit).  On Espressif devkits, both connections are available, and will be labeled.  ESP32 can only use UART, so will only have one connector.  Other variants with one connector will typically be using USB.
```

Comment:
I would recommend adding the _**"Please check in the product product page/datasheet"**_ link to [products.espressif.com](products.espressif.com) for the ESP32-X products with the embedded USB-Serial and JTAG.